djangosaml2idp/views.py

```diff
@@ -144,16 +185,33 @@ def get(self, request, *args, **kwargs):
 
         # Construct SamlResponse message
         try:
-            authn_resp = self.IDP.create_authn_response(
-                identity=identity, userid=request.user.username,
-                name_id=NameID(format=resp_args['name_id_policy'].format, sp_name_qualifier=resp_args['destination'], text=request.user.username),
+            authn_resp_xml = self.IDP.create_authn_response(
+                identity=identity, userid=getattr(request.user, user_identity),
+                name_id=NameID(format=resp_args['name_id_policy'].format, sp_name_qualifier=resp_args['destination'], text=getattr(
+                    request.user, user_identity)),
                 authn=AUTHN_BROKER.get_authn_by_accr(req_authn_context),
-                sign_response=self.IDP.config.getattr("sign_response", "idp") or False,
-                sign_assertion=self.IDP.config.getattr("sign_assertion", "idp") or False,
+                sign_response=self.IDP.config.getattr(
+                    "sign_response", "idp") or False,
+                sign_assertion=self.IDP.config.getattr(
+                    "sign_assertion", "idp") or False,
                 **resp_args)
         except Exception as excp:
             return self.handle_error(request, exception=excp, status=500)
 
+        if resp_args['binding'] == BINDING_HTTP_POST:
+            if PY3:
```

Comment:
Current version of the package does already not support Python 2 anymore, so you can omit the if/else and just use the 3 case.